SConstruct

```diff
@@ -282,11 +282,14 @@ opts.Add(BoolVariable("builtin_zstd", "Use the built-in Zstd library", True))
 opts.Add("CXX", "C++ compiler binary")
 opts.Add("CC", "C compiler binary")
 opts.Add("LINK", "Linker binary")
```

Comment:
Should we also add options for `AS`, `AR`, `RANLIB`, `RC`, and maybe `S_compiler`?

Is there maybe a less hacky way to achieve this with builtin SCons features? CC @Repiteo 

See related #91743.

Reply:
To clarify, exposing the `*flags` as done in this PR should be fine to merge already, as this will work fine.

Making it possible to actually override the binary for each step reliably will require more work and an approach similar to #91743, so we can leave that to discuss there.

Reply:
> Should we also add options for AS, AR, RANLIB, RC, and maybe S_compiler?

Why not? I have not tested https://github.com/godotengine/godot/pull/91743, but it seems fine to me (at least the idea).